### PR TITLE
chore: add frameworkmetadata and surrounding unit tests

### PR DESCRIPTION
**Pull request by wooj2**
Prior to figuring out how to expose this for our customers, it seems like this is heading in the right direction, so i wanted to:
* Gain consensus that this is the correct direction
* add surrounding unit tests so we make sure we aren't breaking anything

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AWSClientRuntime/Tests/Http/AWSUserAgentMetadataTests.swift [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+@testable import AWSClientRuntime
+
+class AWSUseragentMetadataTests: XCTestCase {
```
Comment (kneekey23) [inline AWSClientRuntime/Tests/Http/AWSUserAgentMetadataTests.swift:11]: were you not able to find the existing user agent tests?

File: AWSClientRuntime/Tests/Http/FrameworkMetadataTests.swift [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                      "test2": "SomethingOtherThanANumber"]
+
+        let sut = FrameworkMetadata(name: "aws-amplify", version: "2.0.0", extras: extras)
+        
+        let option1 = "lib/aws-amplify/2.0.0 md/test1/1.0 md/test2/SomethingOtherThanANumber" == sut.description
+        let option2 = "lib/aws-amplify/2.0.0 md/test2/SomethingOtherThanANumber md/test1/1.0" == sut.description
+        XCTAssert(option1 || option2)
+    }
+
+    func testMetadataIsSanitized() {
```
Comment (kneekey23) [inline AWSClientRuntime/Tests/Http/FrameworkMetadataTests.swift:43]: nice test!

**Review by kneekey23 — APPROVED** [review]
LGTM, just had one question about testing but definitely in the right direction!

File: AWSClientRuntime/Tests/Http/AWSUserAgentMetadataTests.swift (hunk 1 above)
Comment (wooj2) [reply AWSClientRuntime/Tests/Http/AWSUserAgentMetadataTests.swift:11]: Ah, I didn't notice there were already UserAgentTests inside of a folder called "UserAgent" -- considering the source files were located in a folder called `Http` I was expecting the accompanying unit tests would be there as well -- i'll move the test over. to be consistent

**Pull request merged**